secp256k1.nim

```diff
@@ -517,3 +517,7 @@ proc default*(T: type SkSignature): T {.error: "loophole".}
 proc default*(T: type SkRecoverableSignature): T {.error: "loophole".}
 proc default*(T: type SkEcdhSecret): T {.error: "loophole".}
 proc default*(T: type SkEcdhRawSecret): T {.error: "loophole".}
+
+func privKeyTweakAdd*(secretKey: var openArray[byte], tweak: openArray[byte]) =
+  let res = secp256k1_ec_privkey_tweak_add(getContext(), secretKey.ptr0, tweak.ptr0)
+  doAssert res == 1
```

Comment:
I guess it's tolerable to have this assert here, but please add a comment explaining when it will trigger:

From the secp256k1 docs:

```nim
# the tweak was out of range (chance of around 1 in 2^128 for uniformly random 32-byte arrays,
# or the resulting private key is invalid (only when the tweak is the complement of the private key
```

Reply:
hm - good catch - in this case, I think it makes sense to return an error and not use `doAssert`, because of how error reporting in libsecp works with a callback that may abort the process etc - the tweak in this case is user-supplied and hasn't been validated - the other `doAssert` in this file are safe because `SkSecretKey` and friends are checked on construction